src/pnpq/devices/switch_thorlabs_osw_e.py

```diff
@@ -213,32 +224,103 @@ def get_state(self) -> State:
             return self._get_state()
 
     def _get_state(self) -> State:
-        """Private method to get the status of the switch without locks. This is used to check the status during set_state."""
+        """Private method to get the status of the switch without locks. This is used to check the status during set_state.
+
+        :raises ThorlabsOswError: If the device reports an error or sends an unexpected reply that cannot be parsed as a valid state.
+        """
         command = b"S?\n"
         self._connection.write(command)
         response = self._read_serial_response()
-        return State(int(response.decode("utf-8")))
+        try:
+            value_str = response.decode("utf-8")
+            value_int = int(value_str)
+            return State(value_int)
```

Comment:
I'm not sure why we've taken a single line of code and rewritten it as three lines.